### ISSUE TITLE
Correctly sign gradle-download-task pom files

### ISSUE THREAD

**Issue by liutikas** [michel-kraemer/gradle-download-task#187]
If you try to enable [Gradle signature verification](https://docs.gradle.org/current/userguide/dependency_verification.html#sec:signature-verification) when using `de.undercouch:gradle-download-task:4.1.2` you get two failures:

1. `gradle-download-task-4.1.2.pom` is signed incorrectly. `Artifact was signed with key 1fa37fbe4453c1073e7ef61d6449005f96bc97a3 but signature didn't match`
2. `de.undercouch.download.gradle.plugin-4.1.2.pom` is not signed at all

To repro:
1. Add `plugins { id("de.undercouch.download") version "4.1.2" }` to your build.gradle.kts
2. Add `gradle/verification-metadata.xml` file with:
```
<?xml version="1.0" encoding="UTF-8"?>
<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.1.xsd">
   <configuration>
      <verify-metadata>true</verify-metadata>
      <verify-signatures>true</verify-signatures>
      <trusted-keys>
         <trusted-key id="19beab2d799c020f17c69126b16698a4adf4d638" group="org.checkerframework"/>
         <trusted-key id="7615ad56144df2376f49d98b1669c4bb543e0445" group="com.google.errorprone"/>
         <trusted-key id="7616eb882daf57a11477aaf559a252fb1199d873" group="com.google.code.findbugs"/>
         <trusted-key id="694621a7227d8d5289699830abe9f3126bb741c1" group="com.google.guava"/>
         <trusted-key id="1fa37fbe4453c1073e7ef61d6449005f96bc97a3" group="de.undercouch"/>
         <trusted-key id="afcc4c7594d09e2182c60e0f7a01b0f236e5430f" group="com.google.code.gson"/>
         <trusted-key id="bdb5fa4fe719d787fb3d3197f6d4a1d411e9d1ae" group="com.google.guava"/>
         <trusted-key id="fa7929f83ad44c4590f6cc6815c71c0a4e0b8edd" group="net.java.dev.jna"/>
         <trusted-key id="6f538074ccebf35f28af9b066a0975f8b1127b83" group="org.jetbrains.kotlin"/>
         <trusted-key id="b801e2f8ef035068ec1139cc29579f18fa8fd93b" group="com.google.j2objc"/>
         <trusted-key id="e7dc75fc24fb3c8dfe8086ad3d5839a2262cbbfb" group="org.jetbrains.kotlinx"/>
         <trusted-key id="44fbdbbc1a00fe414f1c1873586654072ead6677" group="org.sonatype.oss"/>
         <trusted-key id="55e770230e69cc6de143fb5b62c82e50836eb3ee" group="com.github.gundy"/>
         <trusted-key id="8756c4f765c9ac3cb6b85d62379ce192d401ab61" group="org.jetbrains.intellij.deps"/>
      </trusted-keys>
   </configuration>
</verification-metadata>
```

This should succeed, but it fails with failures listed above.

**Comment by michel-kraemer** [comment]
gradle-download-task 4.1.2 was published with a very old version of the [com.gradle.plugin-publish](https://plugins.gradle.org/plugin/com.gradle.plugin-publish) plugin. I suppose this is the culprit. gradle-download-task is scheduled to be released this weekend. You should try again with this version.

In the meantime, you can use the version from Maven central instead of the plugin portal. Remove `plugins { id("de.undercouch.download") version "4.1.2" }` from your build script and add the following instead:

```gradle
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath 'de.undercouch:gradle-download-task:4.1.2'
    }
}

apply plugin: 'de.undercouch.download'
```

The artifacts on Maven central should be correctly signed.

**Comment by michel-kraemer** [comment]
Version 5.0.0 has just been released. May I ask you to update and check if the problem persists?

**Comment by liutikas** [comment]
It is still not valid on 5.0.1

For `gradle-download-task-5.0.1.pom` it fails with:
```
Artifact was signed with key b8cba850ddc1adac674ff8e1d3dbc823be4819ed but signature didn't match
```

It also looks like [de.undercouch.download.gradle.plugin-5.0.1.pom](https://plugins.gradle.org/m2/de/undercouch/download/de.undercouch.download.gradle.plugin/5.0.1/) was not signed at all (there is no .asc file) so it cannot be validated via signature check.

**Comment by michel-kraemer** [comment]
OK. I thought upgrading the [com.gradle.plugin-publish](https://plugins.gradle.org/plugin/com.gradle.plugin-publish) plugin would fix the problem, but it didn't. I need to investigate and will get back to you.

**Comment by michel-kraemer** [comment]
I'm sorry but I didn't find a way to solve this issue. I'm not sure if Gradle plugins can be signed at all. I didn't find any other plugin in the repository under https://plugins.gradle.org/m2/ that is signed. Are you sure this is possible?

If you really need this feature, you can define a `pluginMangement` block in your `settings.gradle` and change the source for plugins to Maven Central. The files here are all correctly signed:
https://repo1.maven.org/maven2/de/undercouch/gradle-download-task/5.0.1/

Have a look at the [examples/settings.gradle](https://github.com/michel-kraemer/gradle-download-task/blob/master/examples/settings.gradle) file for more information.

Let me know if this helps and if so please close this issue. Thanks!

**Comment by liutikas** [comment]
I reached out to Gradle folks and you are right that Gradle plugin portal does not allow signing poms. I have created an issue on their side https://github.com/gradle/gradle/issues/20349, once that it solved, you should be able to sign it.

**Comment by michel-kraemer** [comment]
Thanks for reporting this to the Gradle guys! I've pinned this issue so it won't be closed by the stale bot. As soon as the new publish plugin has been released we can close it.

**Comment by liutikas** [comment]
https://plugins.gradle.org/plugin/com.gradle.plugin-publish the latest 1.0.0-rc-1 claims to support signing. I'd love to see if that works for you.

**Comment by michel-kraemer** [comment]
I've released a new version 5.0.5 using the latest version of the plugin-publish plugin. Artifacts in the Gradle plugin portal should now be correctly signed.

Thanks for the pointer to the new plugin-publish plugin!
